test/src/module/command/restoreTest.c

```diff
@@ -3229,14 +3256,12 @@ testRun(void)
 
         HRN_STORAGE_PUT(storagePgWrite(), PG_PATH_BASE "/1/2", relation);
 
-        argList = strLstNew();
-        hrnCfgArgRawZ(argList, cfgOptStanza, "test1");
-        hrnCfgArgRaw(argList, cfgOptRepoPath, repoPath);
-        hrnCfgArgRaw(argList, cfgOptPgPath, pgPath);
-        hrnCfgArgRawZ(argList, cfgOptSpoolPath, TEST_PATH "/spool");
-        hrnCfgArgRawBool(argList, cfgOptDelta, true);
-        hrnCfgArgRawZ(argList, cfgOptRepoCipherType, "aes-256-cbc");
-        hrnCfgEnvRawZ(cfgOptRepoCipherPass, TEST_CIPHER_PASS);
+        // For this restore to succeed we need to set limit time using the timestamp on backup.info
+        String *limitTime = strNewTimeP(
+            "%Y-%m-%d %H:%M:%S+00", storageInfoP(storageRepo(), INFO_BACKUP_PATH_FILE_STR).timeModified, .utc = true);
```

Comment:
Any concerns about sub-second timing here and missing any files changed in the floor() of the second? (i.e., do we need second + 1 or something else involving milli-/microseconds here?)

Reply:
We don't support subsecond timing since not all storage supports it. Added a clarification in c1176f227a3d.

Reply:
Hm, I've got some troubles to understand the clarification:

> Note comparisons to the storage timestamp are &lt;= the timestamp provided and milliseconds are truncated from the timestamp when provided.

Maybe I'm struggling with the grammar or punctuation, i.e. `Note: comparisons...` or `Note that comparisons...` would feel more natural to me :thinking:

Reply:
So the `backup.info` file is the last thing written, so worst case it wouldn't see this as a valid backup (so would skip restoring) if you selected the timestamp that you completed at with partial milliseconds truncated—i.e., nothing inconsistent. 

I do wonder if a little stronger wording in the docs might be warranted, since if you saw a backup with milliseconds attached to the timestamp you'd need to choose a time at least a second after this, but maybe it's a similar issue with other PITR-related things so not worth stressing about.

Reply:
> Maybe I'm struggling with the grammar or punctuation

Definitely a typo on my part. Fixed in c8d450bd109a.

Reply:
> so worst case it wouldn't see this as a valid backup (so would skip restoring) if you selected the timestamp that you completed at with partial milliseconds truncated—i.e., nothing inconsistent.

You would see it since we in effect round up. But yeah, in general either the backup will all be there or not be in backup.info.

> since if you saw a backup with milliseconds attached to the timestamp

Reply:
Ok so if we in effect round up not down I think my concern here is moot, but thanks for considering it. :)